erpnext/accounts/utils.py

```diff
@@ -1847,10 +1847,16 @@ def get_account_type(account):
 					dr_or_cr *= -1
 					dr_or_cr_account_currency *= -1
 
-				against_voucher_type = (
-					gle.against_voucher_type if gle.against_voucher_type else gle.voucher_type
-				)
-				against_voucher_no = gle.against_voucher if gle.against_voucher else gle.voucher_no
+				against_voucher_type = gle.against_voucher_type
+				against_voucher_no = gle.against_voucher
+
+				if gle.against_voucher:
+					if gle.against_voucher == gle.voucher_no and gle.advance_voucher_no:
+						against_voucher_type = gle.advance_voucher_type
+						against_voucher_no = gle.advance_voucher_no
```

Comment:
Unclear on this change. GL Entry doesn't have advance_voucher_type and advance_voucher_no fields